gammapy/irf/tests/test_background.py

```diff
@@ -49,3 +54,58 @@ def test_background_3d_write(bkg_3d):
     hdu = table_to_fits_table(bkg_3d.to_table())
     assert_equal(hdu.data['DETX_LO'][0], bkg_3d.data.axis('detx').lo.value)
     assert hdu.header['TUNIT1'] == bkg_3d.data.axis('detx').lo.unit
+
+
+def make_test_array():
+    # Create a dummy `Background2D`
+    energy = [1, 10, 100] * u.TeV
+    offset = [0, 1, 2, 3] * u.deg
+    data = np.zeros((len(energy)-1,len(offset)-1)) * u.Unit('s-1 MeV-1 sr-1')
+    data.value[1,0]=2
+    data.value[1,1]=4
+    return Background2D(
+    energy_lo=energy[:-1], energy_hi=energy[1:],
+    offset_lo=offset[:-1], offset_hi=offset[1:],
+    data=data,
+)
+
+
+def test_background2d_read_write(tmpdir):
+    bkg_2d_1 = make_test_array()
+    filename = str(tmpdir / "bkg2d.fits")
+    prim_hdu = fits.PrimaryHDU()
+    hdu_bkg = bkg_2d_1.to_fits()
+    hdulist = fits.HDUList([prim_hdu, hdu_bkg])
+    hdulist.writeto(filename)
+    bkg_2d_2 = Background2D.read(filename)
+
+    axis = bkg_2d_2.data.axis('energy')
+    assert axis.nbins == 2
+    assert axis.unit == 'TeV'
+
+    axis = bkg_2d_2.data.axis('offset')
+    assert axis.nbins == 3
+    assert axis.unit == 'deg'
+
+    data = bkg_2d_2.data.data
+    assert data.shape == (2, 3)
+    assert data.unit == u.Unit('s-1 MeV-1 sr-1')
+
+
+@requires_dependency('scipy')
+def test_background2d_evaluate():
+    bkg_2d = make_test_array()
+    data_unit = u.Unit('s-1 MeV-1 sr-1')
+
+    # Interoplate at the energy and offset bin of the bgk_2d axes
+    off = bkg_2d.data.axis('offset').nodes
+    e_reco = bkg_2d.data.axis('energy').nodes
+    res = bkg_2d.evaluate(fov_offset=off, energy_reco=e_reco)
+    assert_quantity_allclose(res, bkg_2d.data.data)
+
+    #Test linear interpolation for offset
```

Comment:
@cdeil 
This is easy to to a easy test example regarding the offset. But since the nodes will be the log center for the energy then I can not use just integer anymore

Reply:
I think I would manually compute `sqrt(e_1 * e_2)`, i.e. log center, and then use that in the evaluate call, see if it's giving the mean as value between the IRF value at the two nodes in one case.